docs/design.md

```diff
@@ -0,0 +1,47 @@
+# Cluster API Provider Azure Design<!-- omit in toc -->
+
+![Design](./design.jpg)
+
+## Folder Structure
```

Comment:
Let's remove this; we'll have to keep track of it every time something changes.

Reply:
Yes, we should keep track of the design changes, and any future reviewers would be able to just refer this instead of unaware of why it was done a certain way. I prefer to keep this for now